test/retcodes_test.py

```diff
@@ -170,3 +171,68 @@ def new_func(*args, **kwargs):
         with mock.patch('luigi.scheduler.Scheduler.add_task', new_func):
             self.run_and_expect('RequiringTask', 0)
             self.run_and_expect('RequiringTask --retcode-not-run 5', 5)
+
+class RetcodesWithTaskRetryTest(LuigiTestCase):
```

Comment:
About this, `run_with_config` was the critical point to create a separate class, because the existing one only handle retcode configuration but i want to pass `retry-delay`.

I think these two class can merge by changing the existing side `run_with_config`.
The file system and sleep trick are needed similar to `execution_summary_test`...